File: 5 Testen und Debugging/(1) Digitaluhr_V1/Nummernanzeige.java
/**
 * Die Klasse Nummernanzeige simuliert eine zweistellige Anzeige,
 * ... im Bereich von 0 bis limit.
 * Autor:   Martin Lentz
 * Version: 01.10.2014
 */
public class Nummernanzeige
{
    private int wert, limit;

    public Nummernanzeige(int limit)
    {
        setzeWert(0);
        this.limit = limit;
    }

    // Lese aktuellen Wert aus Nummernanzeige aus
    public int gibWert()
    {
        return wert;
    }
    
    // Setze Nummernanzeige auf neuen Wert
    public void setzeWert(int wert)
    {
        if ( (wert <= limit) && (wert >= 0) )
        {
            this.wert = wert;
        }
    }
    
    // Erhöhe aktuellen Wert der Nummernanzeige
    public void erhoeheWert()
    {
        wert = wert+1;
        if (wert > limit)
        {
            wert = 0;
        }
    }
}

File: Informatik/2 C#-Grundlagen und GUIs/8 Digitaluhr_V2/Digitaluhr_V2/Nummernanzeige.cs
namespace Digitaluhr_V2
{
    public class Nummernanzeige
    {
        private readonly int limit;
        private int wert;

        public Nummernanzeige(int limit)
        {
            Wert = 0;
            this.limit = limit;
        }

        public int Wert
        {
            get { return wert; }
            set
            {
                if (value < 0 || value > limit) wert = 0;
                else wert = value;
            }
        }

        public void ErhoeheWert()
        {
            Wert++;
        }

        public override string ToString()
        {
            return Wert.ToString("00");
        }
    }
}